### PR TITLE
[perf](Nereids) add back canEliminate temporary

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run clickbench

**Review by englefly — APPROVED** [review]

File: regression-test/suites/nereids_rules_p0/eager_aggregate/basic.groovy [hunk 1]
```diff
@@ -18,6 +18,7 @@
 suite("eager_aggregate_basic") {
     sql "SET enable_nereids_planner=true"
     sql "SET enable_fallback_to_original_planner=false"
+    sql "SET ignore_shape_nodes='PhysicalDistribute,PhysicalProject'"
```
Comment (xzj7019) [inline regression-test/suites/nereids_rules_p0/eager_aggregate/basic.groovy:21]: need to confirm

**Review by xzj7019 — APPROVED** [review]

**Pull request merged**